ui/src/app/components/workflows/workflow/dynamic-parts/cron-quartz-part/cron-quartz-part.component.html

```diff
@@ -1,72 +1,101 @@
+              <option *ngFor="let option of Frequencies | keyvalue" [selected]="frequency==option.value" [value]="option.value">
+                {{option.value}}</option>
+            </select>
+          </div>
 
-    <span *ngIf="base >= 3">at</span>
+        <span *ngIf="frequency == Frequencies.DAY">at</span>
 
-    <div class="clr-select-wrapper" *ngIf="base >= 3">
-      <select class="clr-select" clrControl [(ngModel)]="everyHourAt">
-        <option *ngFor="let option of hourValues" [value]="option">{{option}}</option>
-      </select>
-    </div>
+        <div class="clr-select-wrapper" *ngIf="frequency == Frequencies.DAY">
+          <select class="clr-select" clrControl [ngModel]="day" (ngModelChange)="onDayChange($event)">
+            <option *ngFor="let option of DayValues" [selected]="day==option" [value]="option">{{option}}</option>
+          </select>
+        </div>
 
-    <span *ngIf="base >= 3">:</span>
 
-    <div class="clr-select-wrapper" *ngIf="base >= 2">
-      <select class="clr-select" clrControl [(ngModel)]="minuteValue">
-        <option *ngFor="let option of minuteValues" [value]="option">{{option}}</option>
-      </select>
-    </div>
+        <div class="clr-select-wrapper" *ngIf="frequency == Frequencies.HOUR_AT">
+          <select class="clr-select" clrControl [ngModel]="hourAt" (ngModelChange)="onHourAtChange($event)" >
+            <option *ngFor="let option of HourAtValues" [selected]="hourAt==option" [value]="option">{{option}}</option>
+          </select>
+        </div>
 
-    <div class="clr-select-wrapper" *ngIf="base == 1">
-      <select class="clr-select" clrControl [(ngModel)]="minuteValue">
-        <option *ngFor="let option of minutesValues" [value]="option">{{option}}</option>
-      </select>
-    </div>
+        <div class="clr-select-wrapper" *ngIf="frequency == Frequencies.HOUR_EVERY">
+          <select class="clr-select" clrControl [ngModel]="hourEvery" (ngModelChange)="onHourEveryChange($event)">
+            <option *ngFor="let option of HourEveryValues" [selected]="hourEvery==option" [value]="option">{{option}}</option>
+          </select>
+        </div>
 
-    <span *ngIf="base == 2"> past the hour</span>
+        <span *ngIf="frequency == Frequencies.HOUR_EVERY"> minutes</span>
+        <span *ngIf="frequency == Frequencies.DAY"> o'clock</span>
+        <span *ngIf="frequency == Frequencies.HOUR_AT"> minutes past the hour</span>
 
-    <span *ngIf="base == 1"> minutes</span>
+      </clr-control-container>
 
 
-  </clr-control-container>
+
+
+  </div>
+
 </div>
 
+
 <div *ngIf="isShow">
   <clr-control-container>
-    <label>{{name}}</label>
-    <input clrControl type="text" [readOnly]="true"  name="workflowName" [size]="40" value="Every day at 12:00" />
+    <label>Frequency</label>
+    <input *ngIf="(validation | async) as validationResult; else loading" [readOnly]="true" clrControl
```

Comment:
How about `[size]="100"` to be on the safe side? For sth like `0/10 15 16-22 ? 3-7 * 2020,2022`

Reply:
Fixed